### ISSUE TITLE
Spectrum loader takes up a lot of CPU/GPU cycles

### ISSUE THREAD

**Issue by shmmsra** [adobe/spectrum-css#722]
## Description

Whenever spectrum loader is shown on UI, it takes a lot of CPU/GPU cycles leading to high resource consumption on the machine. It triggers a false alarm and causes a negative impact on the overall performance of the application and machine.

## Steps to reproduce
1. Open a browser e.g., cefclient app
2. Load a page having spectrum loader
3. Observe the CPU consumption for the app goes too high in the activity monitor

## Expected behavior

Showing a loader on the screen should not be so costly from a resource consumption point of view.

## Screenshots

Following is a screenshot with high CPU consumtion when spectrum loaded in shown on screen:
<img width="1792" alt="high-cpu" src="https://user-images.githubusercontent.com/10760114/81719235-6ba3e900-949a-11ea-878c-ffd9482cfd89.png">

Following is a screenshot with normal CPU consumption without spectrum loader on screen:
<img width="1792" alt="normal-cpu" src="https://user-images.githubusercontent.com/10760114/81719222-66469e80-949a-11ea-9d37-b62848e55bcc.png">

## Environment
 - **Spectrum CSS version:** 2.11.0
 - **Browser(s) and OS(s):** Equivalent of Chrome 78.0.3904.87

## Additional context

Original tickets:
https://git.corp.adobe.com/React/react-spectrum-v2/issues/502
https://jira.corp.adobe.com/browse/CCD-8075

**Comment by snowystinger** [comment]
Took a look at this one and tried a few alternatives:
Canvas 2d, ends up having a similar performance profile to animated div's when you have many. If you only have a couple, it's pretty good.  https://jsfiddle.net/snowystinger/2eacszho/
Note: this draws each variant once and copies that data to each canvas.
```
With 1 canvas, the chrome processes in Activity Monitor report 14% CPU, 0.8% GPU. With 24 canvas (8x set of 3 unique) it’s 70% CPU, 5% GPU, With 48 canvas (16x set of 3 unique), it’s 95% CPU, 8.4% GPU.
CSS animation of regular div’s was, for 24 (8x of the 3 sizes), 80% CPU, 15% GPU.  For 48 70% CPU, 45% GPU.
```
Note, this is just chrome. On Safari, the two perform about the same.

Next I tried webgl. However, from the get go, we're going to be hit with a limitation that differs browser to browser. Number of webgl instances is between 8-16. I didn't go very far down this path once I realized that.

We used to have svg sprite files that we'd animated across, however, that was too big of a file size and especially if you needed multiple variants.

Some other avenue's that could be explored:
- Build the svg sprite sheet locally.
- Use a single SVG definition with svg animation on a dashed border. (we couldn't do this earlier because of IE11)
- Use a webgl instance to draw all the frames and compile it into a gif or series of bitmaps. I tried the bitmap route a little bit in the 2d version, but I'd be worried about memory usage, plus it didn't seem to help much.
- Talking to design about limitations of browsers and devising a different indefinite loader (last resort)
